src/Spatial/Euclidean/StraightLine2D.cs

```diff
@@ -0,0 +1,315 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace MathNet.Spatial.Euclidean
+{
+    /// <summary>
+    /// Represents a line conforming to the equation y = mx + b where m is the gradient and b is the y-intercept
+    /// </summary>
+    public struct StraightLine2D : ILine
+    {
+        private readonly double yoffset;
+        private readonly double xoffset;
+
+        /// <summary>
+        /// The gradient of the line
+        /// </summary>
+        public double Gradient { get; }
+
+        /// <summary>
+        /// Creates a Line conforming to the equation y = mx + b where point is a point on the line and gradient is the slope of the line
+        /// </summary>
+        /// <param name="point">A point on the line</param>
+        /// <param name="gradient">The gradient of the line</param>
+        public StraightLine2D(Point2D point, double gradient)
+        {
+            Gradient = gradient;
+            if (double.IsInfinity(Gradient))
+            {
+                yoffset = double.NaN; // special case vertical line has no y-intercept
+                xoffset = point.X;
+            }
+            else
+            {
+                yoffset = point.Y - (Gradient * point.X);
+                xoffset = double.NaN;
+            }
+        }
+
+        /// <summary>
+        /// Creates a Line conforming to the equation y = mx + b which passes through points p1 and p2
+        /// </summary>
+        /// <param name="p1">A point on the line</param>
+        /// <param name="p2">A point on the line</param>
+        public StraightLine2D(Point2D p1, Point2D p2)
+        {
+            if (p1 == p2)
+            {
+                throw new ArgumentException("A line cannot be formed from two identical points");
+            }
+
+            Gradient = (p2.Y - p1.Y) / (p2.X - p1.X);
+            if (double.IsInfinity(Gradient))
+            {
+                yoffset = double.NaN; // special case vertical line has no y-intercept
+                xoffset = p1.X;
+            }
+            else
+            {
+                yoffset = p1.Y - (Gradient * p1.X);
+                xoffset = double.NaN;
+            }
+        }
+
+        /// <summary>
+        /// Creates a Line conforming to the equation y = mx + b which passes through the points provided
+        /// </summary>
+        /// <param name="points">The points though which the line should pass</param>
+        /// <exception cref="ArgumentException">Thrown when less than 2 points are provided</exception>
+        /// <exception cref="ArgumentException">Thrown when points are not collinear</exception>
+        public StraightLine2D(IEnumerable<Point2D> points) : this(points.ToArray())
```

Comment:
Not sure we should have this ctor, I know you added it for consistency but maybe we should go the other way and deprecate the other places?

Reply:
changing ctors to factory methods Create, CreateFromCollinear